src/main/java/org/truffleruby/language/methods/CachedLazyCallTargetSupplier.java

```diff
@@ -26,6 +26,12 @@ public CachedLazyCallTargetSupplier(Supplier<RootCallTarget> supplier) {
 
     public RootCallTarget get() {
         CompilerAsserts.neverPartOfCompilation("Only behind a transfer, must not PE the Supplier");
+
+        RootCallTarget alreadySet = this.callTarget;
+        if (alreadySet != null) {
+            return alreadySet;
+        }
+
```

Comment:
Maybe we shouldn't use a `CachedLazyCallTargetSupplier` when lazy translation is disabled?

Reply:
Maybe, but out of scope of this PR, and unclear because https://github.com/oracle/truffleruby/blob/b131d3c146afe6bed0f119d0891b34ba68ac5deb/src/main/java/org/truffleruby/parser/MethodTranslator.java#L429-L430
That actually creates the CallTarget lazily, while computing the RootNode eagerly, which can still save some time.
That means even `--lazy-calltargets=false` doesn't do it fully eagerly, there are probably trade-offs there.